src/upnp.cpp

```diff
@@ -955,6 +955,8 @@ void upnp::on_upnp_xml(error_code const& e
 		d.upnp_connection.reset();
 	}
 
+	if (m_closing) return;
```

Comment:
what worries me a bit with the placement of this is that ``d.disabled = true`` may be skipped. Specifically in the case of shutting down and the error may be ``eof``

Reply:
I put it there because `on_upnp_get_ip_address_response` has it in the same place, do you prefer I move both to below the `eof` logic (or remove both)? or just this one?